web/api/db/schema.prisma

```diff
@@ -102,4 +102,21 @@ model CachedProfile {
 
   createdAt DateTime @default(now())
   updatedAt DateTime @updatedAt
+
+  connections Connection[]
+}
+
+
+model Connection {
+  @@id([ profileId, purposeIdentifier ])
```

Comment:
Although both Prisma and Postgres have first-class support for multi-column primary keys, I think it generally makes life easier to use a simple integer ID column instead, and then add a multi-column unique index as well.

A couple concrete examples of where a dedicated primary key works better:
1. GraphQL clients (including apollo-client) are [generally configured](https://graphql.org/learn/global-object-identification/) to use an `id: ID!` field as a global cache key, and automatically update objects within the cache if you send a new query or mutation that returns an object with the same type and ID.
2. If we need to reference the `Connection` table from another model, it would be more convenient to do that by adding a single `connectionId` field vs `connectionProfileId` and `connectionPurposeIdentifier` fields.

The exception would be for pure join tables, where using the foreign keys of the two joined tables as the join table's primary key can make sense.